### PR TITLE
[nested tensor]add split and layer_norm_backward operations

### PR DESCRIPTION
**Pull request by YuqingJ**
Summary:
Add split and layer_norm_backward.

Note: It is non trivial to support split_with_sizes backward so adding the split operation to support the use case in the model.

Test Plan: unit tests

Differential Revision: D51052966

### CODE REVIEW

File: torch/nested/_internal/ops.py [hunk 1]
```diff
@@ -472,6 +472,31 @@ def prod_dim_int(func, *args, **kwargs):
...
+def split_tensor(func, *args, **kwargs):
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+
+    inp = new_kwargs.pop("input")
+    values = inp._values
+
+    # hack to split on the last dim
+    dim = new_kwargs["dim"]
```
Comment (soulitzer) [inline torch/nested/_internal/ops.py:487]: You can use `new_kwargs["dim"] = _wrap_jagged_dim(inp.ndim(), new_kwargs["dim"], "split")` to normalize the dim.

File: torch/nested/_internal/ops.py [hunk 2]
```diff
@@ -472,6 +472,31 @@ def prod_dim_int(func, *args, **kwargs):
...
+    if dim != -1:
+        raise RuntimeError(
+            "split(): only supported for NestedTensor on dim = -1 for now"
+        )
+
+    split_size = new_kwargs["split_size"]
+    return tuple(
+        NestedTensor(values=x, **extract_kwargs(inp))
+        for x in torch.split(values, split_size, -1)
```
Comment (soulitzer) [inline torch/nested/_internal/ops.py:496]:
```suggestion
        for x in func(inp._values, **new_kwargs)
```
nit: no need to do `split_size = new_kwargs["split_size"]` etc, you can just use `new_kwargs` directly

File: test/test_nestedtensor.py [hunk 3]
```diff
@@ -3270,6 +3293,25 @@ def test_unbind(self, device):
             for i, t in enumerate(out):
                 self.assertEqual(t, tensor_list[i])
 
+    def test_layer_norm(self, device):
+        a = torch.randn(2, 3, requires_grad=True, dtype=torch.float64, device=device)
+        b = torch.randn(3, 3, requires_grad=True, dtype=torch.float64, device=device)
+        c = torch.randn(4, 3, requires_grad=True, dtype=torch.float64, device=device)
```
Comment (soulitzer) [inline test/test_nestedtensor.py:3299]: FYI, we recently added this new helper so you can do something like `ts = self._get_list_for_jagged_tensor(((2, 3, 4), 3), device=device, requires_grad=True)`

File: torch/nested/_internal/ops.py [hunk 4]
```diff
@@ -730,6 +755,20 @@ def native_layer_norm_default(func, *args, **kwargs):
...
+    "grad_out: jt, input: jt, normalized_shape: any, mean: any, rstd: any, weight: any?, bias: any?, output_mask: any",
+)
+def native_layer_norm_default(func, *args, **kwargs):
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+    grad_out = new_kwargs.pop("grad_out")
+    inp = new_kwargs.pop("input")
+    d_input, d_gamma, d_beta = func(grad_out._values, inp._values, **new_kwargs)
+    return (NestedTensor(d_input, **extract_kwargs(inp)), d_gamma, d_beta)
```
Comment (soulitzer) [inline torch/nested/_internal/ops.py:769]: We should check if `d_input` is None here in the case where input does not require grad.

**Review by soulitzer — COMMENTED** [review]
Thanks for the addition, mostly minor comments.
One thing to note for backward functions is that they take a `output_mask` which indicates which inputs actually require grad. If a particular input does not require grad, the backward function should return None, and we should handle that specially.

**Review by soulitzer — APPROVED** [review]
Thanks!
